docs/source/hpa.rst

```diff
@@ -0,0 +1,159 @@
+       another-random-sfx-metrics:
+         target_value: 2333
+         signalflow_metrics_query: "data('your_own_sfx_metrics2', filter('some_dimension', 'value')).max(over="1d").publish()"
+       ...
+
+Parameters
+----------
+
+min_replicas (default=0)
+^^^^^^^^^^^^^^^^^^^^^^^^
+The minimum number of replicas(pods) of your service.
+
+max_replicas (required)
+^^^^^^^^^^^^^^^^^^^^^^^
+The maximum number of replicas(pods) of your service.
+
+horizontal_autoscaling
+^^^^^^^^^^^^^^^^^^^^^^^^
+This overrides ``autoscalig``, ``max_instances``, and ``min_instances``.
+
+cpu
+^^^
+Please check here for `algorthm detail <https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details>`_
+
+:target_average_value (required):
+  An Integer from 0 (exclusive) to 100 (inclusive), (0 - 100]. This is 100 times
+  of the average value exposed by your http endpoint.
+
+memory
+^^^^^^
+Same as cpu.
+
+http
+^^^^
+Same as cpu.
+Makes a request on a HTTP endpoint on your service. Expects a JSON-formatted
+dictionary with a ``utilization`` field containing a number between 0 and
+1. Note that this is the same as in Marathon autoscaler. The endpoint is ``/status``
+and cannot be changed as of Paasta v0.93.0.
+
+When ``dimensions`` is not provided, the average values of all HTTP metrics exposed
+by HTTP endpoints of all running pods is compared with ``target_average_value``,
+and current number of running pods to calculate desired number of pods. You can find
+your HTTP metrics and its dimensions on SignalFX.
+
+When ``dimensions`` is provided, the value retrieved from signalfx with
+``data('http', filter('dimension_key', 'dimension_value')).mean(over=15m).publish()``
+is used together with ``target_average_value``, and current number of running pods to
+calculate the desired number of pods. This field exists to make it easier for folks who
+want to do autoscaling across clusters with their existing http metrics. You can achieve
+the same function with your own custom metrics. Any suggestions/demands are welcome.
+
+:target_average_value (required):
+  An Integer from 0 (exclusive) to 100 (inclusive), (0 - 100]. This is 100 times of the average value exposed by your
+  http endpoint.
+
+:dimensions:
+  Any number of custom key value pairs that are strings. Both key and value need to conform to `SignalFX requirement <https://developers.signalfx.com/metrics/metric_data_overview.html#_dimensions_2>`_
+  Also, the dimension needs to conform to Kubernetes metrics name requirement. The regex used
+  to validate is ``[a-z]([-a-z0-9]*[a-z0-9])``.
+  **Please make sure your dimensions are different from the dimensions used others**
+  HPAMetrics Collector will collect metrics from your HTTP endpoints, add your specified
+  dimensions, and send them to SignalFX. So please make sure your service won't cause troubles
+  for other people.
+
+uwsgi
+^^^^^
+ Same as http, but "/status/uwsgi" is used as the endpoint.
+
+your-signalfx-own-metrics
```

Comment:
Make sure this matches the actual YAML provided (`your-own-sfx-metrics` or `another-random-sfx-metrics`).